packages/server/graphql/types/User.ts

```diff
@@ -562,11 +562,27 @@ const User: GraphQLObjectType<any, GQLContext> = new GraphQLObjectType<any, GQLC
       ) => {
         const viewerId = getUserId(authToken)
         const user = (await dataLoader.get('users').load(userId))!
-        const teamIds =
-          viewerId === userId || isSuperUser(authToken)
-            ? user.tms
-            : user.tms.filter((teamId: string) => authToken.tms.includes(teamId))
+        const organizationsForUser = await dataLoader.get('organizationUsersByUserId').load(userId)
+        const orgIds = organizationsForUser.map(({orgId}: OrganizationUserType) => orgId)
+
+        const orgsWithPublicTeam = []
+
+        for (const orgId of orgIds) {
+          const org = await dataLoader.get('organizations').load(orgId)
+          if (org.featureFlags?.includes('publicTeams')) {
+            const tms = await dataLoader.get('teamsByOrgIds').load(org.id)
+            orgsWithPublicTeam.push(...tms.filter((team) => !user.tms.includes(team.id)))
+          }
+        }
+
+        const isViewerOrSuperUser = viewerId === userId || isSuperUser(authToken)
+
+        const teamIds = isViewerOrSuperUser
+          ? user.tms
+          : user.tms.filter((teamId: string) => authToken.tms.includes(teamId))
+
         const teams = (await dataLoader.get('teams').loadMany(teamIds)).filter(isValid)
+        teams.push(...orgsWithPublicTeam)
```

Comment:
+1 `teams` are all of the teams the user belongs to, whereas `orgsWithPublicTeam` are all of the teams in the orgs that the user belongs to.

This wasn't immediately obvious to me - perhaps the variable names could be updated to make this clearer? I'd also rather have a final variable called something like `const allVisibleTeams = [...teams, ...orgsWithPublicTeam]` rather than using `.push`.

Reply:
I like it! I'd be tempted to change the name of `teams` too to signal the difference, e.g. `teamsUserBelongsTo`, but I'm probably being too pedantic about naming